src/Editor/Util/AssetConverterSkeleton.cpp

```diff
@@ -116,19 +150,55 @@ const std::string& AssetConverterSkeleton::GetErrorString() {
     return errorString;
 }
 
-void AssetConverterSkeleton::BoneRecursive(aiNode* node, int parent) {
-    for (unsigned int i = 0; i < node->mNumChildren; ++i) {
+void AssetConverterSkeleton::SceneRecursive(aiNode* node, int parent) {
+    if (!node)
+        return;
+
+    int thisId = currentId;
+
+    for (auto i = 0; i < node->mNumChildren; ++i) {
         aiNode* child = node->mChildren[i];
+        bool foundBone = false;
+        for (auto j = 0; j < bones.size(); ++j)
+            if (child->mName.C_Str() == bones[j]) {
+                children.push_back(child->mName.C_Str());
+                ids.push_back(0);
+                ++currentId;
+                parents.push_back(-1);
+                BoneRecursive(child, 0);
+            }
 
-        for (unsigned int j = 0; j < bones.size(); ++j) {
+        if (!foundBone)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Condition '!foundBone' is always true [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AknownConditionTrueFalse)